### PR TITLE
Include priority class

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Changes proposed in this pull request:

- Introduce priority class for daemonsets
- Include tests for the priority
- Include tests for logs

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/17157

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ func main() {
 	flag.StringVar(&fluentBitMemoryRequest, "fluent-bit-memory-request", "256Mi", "Memory request for fluent-bit")
 	flag.StringVar(&fluentBitImageVersion, "fluent-bit-image", fluentBitImage, "Image for fluent-bit")
 	flag.StringVar(&fluentBitExporterVersion, "fluent-bit-exporter-image", fluentBitExporterImage, "Image for exporting fluent bit filesystem usage")
-	flag.StringVar(&fluentBitPriorityClassName, "fluent-bit-priority-class-name", "", "Name of the priority class of fluent bit ")
+	flag.StringVar(&fluentBitPriorityClassName, "fluent-bit-priority-class-name", "telemetry-priority-class-high", "Name of the priority class of fluent bit ")
```
Comment (chrkl) [inline main.go:217]: Let's not make this a default to still allow running the operator for local development.

File: config/manager/priority_class.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: priority-class
```
Comment (chrkl) [inline config/manager/priority_class.yaml:4]: This class is never applied, right? We should use it for the operator and the otel collectors (trace-collector-priority-class and metric-gateway-priority-class)

File: test/e2e/manager_test.go [hunk 3]
```diff
@@ -164,5 +165,17 @@ var _ = Describe("Telemetry-manager", func() {
 			err := k8sClient.Get(ctx, key, &crd)
 			Expect(err).NotTo(HaveOccurred())
 		})
+		It("Should have priority class resource created", func() {
```
Comment (chrkl) [inline test/e2e/manager_test.go:168]: Add an empty line above

File: controllers/telemetry/logpipeline_controller_test.go [hunk 4]
```diff
@@ -308,6 +308,21 @@ var _ = Describe("LogPipeline controller", Ordered, func() {
 				return err
 			}, timeout, interval).Should(BeNil())
 		})
+		It("Should have the correct priority class", func() {
```
Comment (chrkl) [inline controllers/telemetry/logpipeline_controller_test.go:311]: We should test the same for the otel collectors.

Comment (rakesh-garimella) [reply controllers/telemetry/logpipeline_controller_test.go:311]: done

File: controllers/telemetry/logpipeline_controller_test.go [hunk 5]
```diff
@@ -308,6 +308,21 @@ var _ = Describe("LogPipeline controller", Ordered, func() {
...
+				var fluentBitDaemonSet appsv1.DaemonSet
+				err := k8sClient.Get(ctx, types.NamespacedName{
+					Name:      testLogPipelineConfig.DaemonSet.Name,
+					Namespace: testLogPipelineConfig.DaemonSet.Namespace,
+				}, &fluentBitDaemonSet)
+				if err != nil {
+					return err
+				}
+				priorityClassName := fluentBitDaemonSet.Spec.Template.Spec.PriorityClassName
+				Expect(priorityClassName).To(Equal("telemetry-priority-class-high"))
```
[the remark this answers is not in the document]
Comment (rakesh-garimella) [reply controllers/telemetry/logpipeline_controller_test.go]: done

**Pull request merged**